### PR TITLE
Fixes #401: Opencti analyzer

### LINKED ISSUE #401: OpenCTI analyzer
We could create an analyzer for [OpenCTI](https://www.opencti.io/en/). This would leverage the [official python client](https://github.com/OpenCTI-Platform/client-python) to look for a specific observable in a OpentCTI instance.

This would require some knowledge of OpenCTI and set up an OpenCTI instance locally. We could use [how it is implemented in Cortex](https://github.com/TheHive-Project/Cortex-Analyzers/blob/master/analyzers/OpenCTI/opencti.py) as an example

### PR DESCRIPTION
**Pull request by sp35**
# Description
This adds an observable analyzer that searches for observables on a connected OpenCTI instance and returns all linked reports.

Notes
- `requests` was downgraded to `2.25.1` for `pycti==4.5.5`

## Related issues
#401 

## Type of change

- [x] New feature (non-breaking change which adds functionality).

# Checklist

- [x] The pull request is for the branch develop
- [x] If I added a new analyzer, I updated the file [Usage](https://github.com/intelowlproject/IntelOwl/blob/master/docs/source/Usage.md). If the analyzer provides additional optional configuration, I added the available options here: [Advanced-Usage](./Advanced-Usage.md)
- [x] If I added external libraries/packages that use restrictive licenses, please add them in the [ReadMe - Legal Notice](https://github.com/certego/IntelOwl/blob/master/README.md) section
- [x] I added new secrets in the files [env_file_app_template](https://github.com/intelowlproject/IntelOwl/blob/master/docker/env_file_app_template), [env_file_app_ci](https://github.com/certego/IntelOwl/blob/master/docker/env_file_app_ci) and in the docs: [Installation](./Installation.md)
- [x] I have added tests
- [ ] The tests gave 0 errors.
- [x] `Black` gave 0 errors.
- [x] `Flake` gave 0 errors.
- [ ] I squashed the commits into a single one. (optional, they will be squashed anyway by the maintainer)
 
# Real World Example
image

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -34,11 +34,12 @@ pysafebrowsing==0.1.1
 PySocks==1.7.1
 python-magic==0.4.22
 quark-engine==21.7.2
-requests==2.26.0
+requests==2.25.1
```
[the remark this answers is not in the document]
Comment (sp35) [reply requirements.txt:37]: Yep, I'll update the description.

File: api_app/analyzers_manager/observable_analyzers/opencti.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
+# This file is a part of IntelOwl https://github.com/intelowlproject/IntelOwl
+# See the file 'LICENSE' for copying permission.
+
+from pycti.api.opencti_api_client import File
+import pycti
+
+from tests.mock_utils import patch, if_mock_connections
+from ..classes import ObservableAnalyzer
```
Comment (Eshaan7) [inline api_app/analyzers_manager/observable_analyzers/opencti.py:8]: nitpick but: use absolute import.

**Comment by Eshaan7** [comment]
Can you see why the tests failed ? Maybe something to do with how mocking was applied. Since the `pycti` imports are on the top level, you can also use the `patch.object()` method.

File: api_app/analyzers_manager/observable_analyzers/opencti.py [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+    @classmethod
+    def _monkeypatch(cls):
+        mock_result = [{"id": 1, "observable_value": "8.8.8.8"}]
+
+        patches = [
+            if_mock_connections(
+                patch("pycti.OpenCTIApiClient", return_value=None),
+                patch("pycti.StixCyberObservable.list", return_value=mock_result),
+                patch("pycti.Report.list", return_value=mock_result),
```
Comment (Eshaan7) [inline api_app/analyzers_manager/observable_analyzers/opencti.py:92]: The circular reference could be a result of this. Define different `return_value` here.

Comment (sp35) [reply api_app/analyzers_manager/observable_analyzers/opencti.py:92]: thank you so much

**Comment by Eshaan7** [comment]
Please rebase this with latest changes from develop-2.

**Comment by Eshaan7** [comment]
Please rebase this with latest changes from develop-2.

**Pull request merged**